src/diff/changes/apiChanges.ts

```diff
@@ -76,4 +84,75 @@ export class ApiChanges {
   getIgnoredChangesCount(): number {
     return this.getChangeCountByCategory(RuleCategory.IGNORED);
   }
+
+  /**
+   * Gets the number of changes in each category
+   */
+  getCategorizedChangeSummary(): Record<RuleCategory, number> {
+    const base = Object.values(RuleCategory).map((c) => [c, 0]);
+    const summaries = this.nodeChanges.map((node) => {
+      return node.getCategorizedChangeSummary();
+    });
+    return _.mergeWith(
+      _.fromPairs(base),
+      ...summaries,
+      (a: number | undefined, b: number): number => (a || 0) + b
+    );
+  }
+
+  /**
+   * Format the changes as a string for printing to console
+   *
+   * @param extraIndent - Number of levels the headings should be nested
+   */
+  toString(extraIndent = 0): string {
```

Comment:
I feel formatting logic should be separated from the data.  This is not a simple toString but a formatting feature. As mentioned before we need a ConsoleFormatter class with an interface so that we can add other formatters (like html) in future

Reply:
I completely agree and was going to make the same comment.